haystack/document_stores/opensearch.py

```diff
@@ -578,15 +719,21 @@ def _validate_and_adjust_document_index(self, index_name: str, headers: Optional
                         f" - Use another embedding field name by setting `embedding_field='my_embedding_field_name'`. "
                     )
 
+                # We use score_script for IVF indices that are not trained yet
+                if self.index_type in ["ivf", "ivf_pq"] and "model_id" not in existing_embedding_field:
+                    knn_engine = "score_script"
+                else:
+                    knn_engine = self.knn_engine
                 # Check if existing embedding field fits desired knn settings
-                if self.knn_engine != "score_script":
+                if knn_engine != "score_script":
```

Comment:
The comment above could be a bit misleading here: we are not actually using score_script.
Should be easier to understand:
```suggestion
                training_required = self.index_type in ["ivf", "ivf_pq"] and "model_id" not in existing_embedding_field
                if self.knn_engine != "score_script" and not training_required:
```